From 76f8d46b0f3f9ba3ad57275bbe0acfcb6d4449f5 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 07 Jan 2024 11:51:36 +0200
Subject: [PATCH] [SW-169995] efuse: do not init efuse implicitly

efuse is initialized explicitly from the main function.
this patch removes the implicit initialization of the efuse from the
hl_fuse_read() because in complex scenarioas such as in multi-die
devices this might create undesired recursion.

Change-Id: I4bd5830ec9f90850b9c014bb363dc50cb918b447
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index fd13f75..8cf6cc8 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -61,8 +61,10 @@
 		else
 			break;
 
-	if (i == FUSE_RETRIES)
+	if (i == FUSE_RETRIES) {
+		fuse_notify_init_fail();
 		return -2;
+	}
 
 	for (i = 0; i < FUSE_WORDS_PER_BANK; i++) {
 		/* read value from bank0 */
@@ -92,18 +94,12 @@
 
 __weak int hl_fuse_read(u32 efuse_base, u32 word, u32 *val)
 {
-	int offset, ret;
+	int offset;
 	u32 read_val;
 
 	if (word >= efuse.words_per_bank) {
 		fuse_notify_invalid_offset();
 		return -1;
-	}
-
-	ret = preboot_fuse_init();
-	if (ret == -2) {
-		fuse_notify_init_fail();
-		return ret;
 	}
 
 	/* read value from bank0 */
